src/payment_channel_management_strategies/PaymentChannelManagementStrategy.js

```diff
@@ -0,0 +1,46 @@
+import { find } from 'lodash';
+
+export default class PaymentChannelManagementStrategy {
+  constructor(sdkContext) {
+    this._sdkContext = sdkContext;
+  }
+
+  async selectChannel(serviceClient) {
+    const account = this._sdkContext.account;
+    await serviceClient.loadOpenChannels();
+    await serviceClient.updateChannelStates();
+    const paymentChannels = serviceClient.paymentChannels;
+    const serviceCallPrice = serviceClient.metadata.pricing.price_in_cogs;
+    const mpeBalance = await account.escrowBalance();
+    const defaultExpiration = await serviceClient.defaultChannelExpiration();
+
+    if(paymentChannels.length === 0) {
+      if(mpeBalance >= serviceCallPrice) {
+        return serviceClient.openChannel(serviceCallPrice, defaultExpiration);
+      }
+
+      return serviceClient.depositAndOpenChannel(serviceCallPrice, defaultExpiration);
+    }
+
+    const firstFundedValidChannel = find(paymentChannels, (paymentChanel) => paymentChanel.hasSufficientFunds(serviceCallPrice) && paymentChanel.isValid(defaultExpiration));
```

Comment:
Service channels should be filtered by `groupId` as well

Reply:
We are filtering the channels while fetching all the open channels [here](https://github.com/singnet/snet-sdk-js/pull/1/files#diff-0e7c20456d55f6ddea0245e78ecb692dR89). And since we are getting the channels from the [serviceClient](https://github.com/singnet/snet-sdk-js/pull/1/files#diff-2651305bb27c2d38666838e4be87c354R12) that would be taken care of. We might have to filter the `PaymentChannels` when we don't get them from `ServiceClient`.

Reply:
It means that channel filtering logic is implemented in two places instead of one. I would move it to single place.

Reply:
The first level of filtering is at the contract level while getting the open channels form the blockchain [here](https://github.com/singnet/snet-sdk-js/pull/1/files#diff-0e7c20456d55f6ddea0245e78ecb692dR83). Which means we will only have channels which are valid for the given `serviceClient`. If we don't do this what will end up happening is that we will get all the `PaymentChannels` between the `service` and the `account` across the group from the blockchain.

The second level of filtering is based on the state of the channel, which needs to be done at the time of `PaymentChannelManagement`.

Reply:
Ok, if this filtering moves part of the work to the Ethereum endpoint then it makes sense. Thank you for explaining.